### DISCUSSION TITLE
3.9.0

### DISCUSSION THREAD

**Discussion by akallabeth** [FreeRDP/FreeRDP#10761] [Q&A]
We're proud to present the newest release of FreeRDP.
 This one brings some major code cleanup (we've addressed lots of clang-tidy
 warnings) as well as some highly anticipated new features.
 We also did update the API documentation quite a bit (still incomplete though,
 help always welcome ;))
 
 So, what is new:
 * Support for RDPEAR (remote credential guard) /remoteGuard option for non windows clients
 * Global configuration file support, allowing to configure certificate
   accept/ignore/... default settings for all users
 * Simplified manpage generation, eliminates docbook and xmlto dependencies
   speeding up builds
 * New API for client channels to run tasks on RDP thread
 * New extended transport layer API
 * RDPECAM MJPEG support 
 * the first updates of timezone definitions from our automated ci
 
 Noteworthy changes:
 * Fix bugs in SSE4.1/AVX2 image copy (#10720)
 * Add warnings for invalid monitor settings during connect (#10672)
 * Fix ALSA microphone support (#10664)
 * Fix modal windows in RAILS mode (#10629)
 * Update experimental SDL3 client (SDL3 API should now have been stabilized,
   various pull requests)
 * Fix keyboard layouts, the external JSON did miss a few (#10639)
 
## New Contributors
* @troyrollo made their first contribution in https://github.com/FreeRDP/FreeRDP/pull/10516
* @JohnWoo22 made their first contribution in https://github.com/FreeRDP/FreeRDP/pull/10662

**Full Changelog**: https://github.com/FreeRDP/FreeRDP/compare/3.7.0...3.9.0

<hr /><em>This discussion was created from the release <a href='https://github.com/FreeRDP/FreeRDP/releases/tag/3.9.0'>3.9.0</a>.</em>

**Comment by VitaliKuzniatsou** [comment]
Hello @akallabeth ,

in 3.8.0 the following run is worked for me:
xfreerdp3 +auto-reconnect +f +fonts -grab-keyboard /multimon /v: /u: /p:

in 3.9.0 I'm getting the following error:
xfreerdp3[07:34:54:343] [14455:00003878] [INFO][com.freerdp.client.x11] - [xf_setup_x11]: Enabling X11 debug mode.
[07:34:54:357] [14455:00003878] [ERROR][com.freerdp.settings] - [find_path_exists_with_dijkstra]: monitor 0 not connected with monitor 2
[07:34:54:357] [14455:00003878] [ERROR][com.freerdp.core] - [freerdp_connect_begin]: ERRCONNECT_PRE_CONNECT_FAILED [0x00020001]
[07:34:54:357] [14455:00003878] [ERROR][com.freerdp.core] - [freerdp_connect_begin]: freerdp_pre_connect failed

Should I attach trace?
Monitor configuration has not been changed,.

Arch Linux, Xorg...

**Comment by akallabeth** [comment]
@VitaliKuzniatsou no trace, but a `xrandr -q` might be of interest.
we´ve added checks for common issues with multimonitor setups.
the RDP version does not allow overlapping areas or monitors that are not touching each other (e.g. a gap between two monitors)

this might be an issue of our `dijkstra` search or your layout is just not really supported (and was previously unchecked)

**Reply by VitaliKuzniatsou** [reply]
@akallabeth please:
xrandr -q
xrandrScreen 0: minimum 320 x 200, current 5040 x 2400, maximum 16384 x 16384
DisplayPort-0 connected primary 1920x1200+3120+1200 (normal left inverted right x axis y axis) 518mm x 324mm
   1920x1200     59.95*+
   1920x1080     60.00  
   1600x1200     60.00  
   1680x1050     59.95  
   1280x1024     60.02  
   1440x900      59.95  
   1280x960      60.00  
   1280x800      59.95  
   1280x720      59.95  
   1024x768      60.00  
   800x600       60.32  
   640x480       59.94  
   720x400       70.08  
DisplayPort-1 connected 1200x1920+1920+463 left (normal left inverted right x axis y axis) 518mm x 324mm
   1920x1200     59.95*+
   1920x1080     60.00  
   1600x1200     60.00  
   1680x1050     59.95  
   1280x1024     60.02  
   1440x900      59.95  
   1280x960      60.00  
   1280x800      59.95  
   1280x720      59.95  
   1024x768      60.00  
   800x600       60.32  
   640x480       59.94  
   720x400       70.08  
HDMI-A-0 connected 1920x1200+0+1080 (normal left inverted right x axis y axis) 518mm x 324mm
   1920x1200     59.95*+
   1920x1080     60.00  
   1600x1200     60.00  
   1680x1050     59.88  
   1280x1024     60.02  
   1440x900      59.95  
   1280x960      60.00  
   1280x800      59.95  
   1280x720      59.95  
   1024x768      60.00  
   800x600       60.32  
   640x480       59.94  
   720x400       70.08  
HDMI-A-1 disconnected (normal left inverted right x axis y axis)
DVI-D-0 connected 1920x1200+3120+0 (normal left inverted right x axis y axis) 518mm x 324mm
   1920x1200     59.95*+
   1920x1080     60.00  
   1600x1200     60.00  
   1680x1050     59.88  
   1280x1024     60.02  
   1440x900      59.95  
   1280x960      60.00  
   1280x800      59.95  
   1280x720      59.95  
   1024x768      60.00  
   800x600       60.32  
   640x480       59.94  
   720x400       70.0

**Reply by akallabeth** [reply]
@VitaliKuzniatsou ok, so you don´t have connections from one monitor to the other.
while this is perfectly valid for `X11` it is not valid for windows.
and `xfreerdp` does just forward the local monitor layout to the remote.

try moving your monitors by one pixel so that you have a one pixel overlap and a path from each monitor to each other monitor and it will work.

**Comment by tigockel** [comment]
> [08:55:47:024] [3058:00000bf4] [ERROR][com.freerdp.settings] - [freerdp_settings_client_monitors_check_primary_and_origin]: Monitor configuration must start at 0/0 for first monitor!

My setup: 2*2560x1440 horizontally aligned with primary on the right. (KDE Plasma may have a gap by default, but since I don't use /multimon this should not be relevent(?))

Cause: [settings.c#L621](https://github.com/FreeRDP/FreeRDP/blob/ae5f655f12397cd18784adfbdf1ed8d1b49c773e/libfreerdp/core/settings.c#L621) and I am not sure if this case should be prevented by default.

Is there a specific reason, that the main monitor must start at 0/0? This restriction did not exist in 3.8.0, since I have not changed my alias. Can I circumvent this in some way?

My alias: `sdl-freerdp3 /u:**** /d:**** /v:localhost:3390 /sec:tls /dynamic-resolution /size:2560x1412 /t:**** +clipboard`

**Reply by akallabeth** [reply]
@tigockel ok, that is an actual bug (should not be enforced in window mode) tracking it here: https://github.com/FreeRDP/FreeRDP/issues/10779

**Reply by tigockel** [reply]
Thanks very much! Right now I change my primary to start the session and immediately change it back. Looking forward for the fix.

**Comment by abappdev** [comment]
I am trying to use FDK AAC ON for android, builds are failing, is this supported for android.

**Reply by akallabeth** [reply]
@abappdev well, a little more details might help.
`fdk-aac` is just another library, so if the headers and library can be detected (e.g. are in `CMAKE_PREFIX_PATH` ) then it works everywhere.

**Reply by abappdev** [reply]
thanks @akallabeth ,here is detailed configuration script

in android-build-freerdp.sh
```
CMAKE_CMD_ARGS="-DANDROID_NDK=$ANDROID_NDK \
	-DANDROID_NATIVE_API_LEVEL=android-${NDK_TARGET} \
	-DCMAKE_TOOLCHAIN_FILE=$ANDROID_NDK/build/cmake/android.toolchain.cmake \
	-DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE \
	-DFREERDP_EXTERNAL_PATH=$BUILD_DST \
	-DWITH_CLIENT_SDL=OFF \
	-DWITH_SERVER=OFF \
	-DWITH_FDK_AAC=ON \
	-DCMAKE_MAKE_PROGRAM=make"
```
	
	
script execution as
```
./scripts/android-build-freerdp.sh --ndk /home/build/Android/Sdk/ndk/25.2.9519653 --sdk /home/build/Android/Sdk --ffmpeg --libssl --libusb --release --openh264 --openh264-ndk /home/build/ndk18.1/android-ndk-r18b --arch $arch
```
faced issue for not found PKGCONFIG, CAIRO, PULSEAUDIO, CJSON, URIPARSER, ALSA such.
nothing modified in cloned repo


we have separately compiled fdk aac so libs for all android arch platforms using toolchain


```
 cd jni/
  ndk-build 
  gedit Android.mk 
  ndk-build 
  gedit Android.mk 
  ndk-build 
  echo $ANDROID_NDK
  gedit ~/.bashrc 
  source ~/.bashrc 
  echo $ANDROID_NDK
  cmake ..   -DCMAKE_TOOLCHAIN_FILE=$ANDROID_NDK/build/cmake/android.toolchain.cmake   -DANDROID_ABI=arm64-v8a   -DANDROID_PLATFORM=android-21   -DCMAKE_BUILD_TYPE=Release
  cmake --build . --config Release
  ./autogen.sh
  ./configure --host=arm-linux-androideabi --enable-static --disable-shared
  make
  cd ..
```

**Reply by akallabeth** [reply]
@abappdev well...
1. for `fdk-aac` you should build with `CMake` *OR* `autotools` - pick one and only one
2. easiest is to install all libraries to a common installation directory you can add to the `CMAKE_PREFIX_PATH` so all `pkg-config` and `CMake` files are automatically detected - so run `cmake --build <builddir> --target install` (you need to set a `cmake -DCMAKE_INSTALL_PREFIX=/tmp/libs/aarch64` or similar during configuration for each arch you want to support)
3. if other dependencies are not found you have to build them - follow https://github.com/FreeRDP/FreeRDP/wiki/Compilation
4. we´re working on replacing out android build shell scripts with https://cmake.org/cmake/help/latest/module/FetchContent.html which might be the right tool for you too

**Reply by 88durgesh** [reply]
I am facing the same issue on the latest Android.
I tried compiling on Ubuntu.
Can you let me know which libraries are required for the CMake install location?
Additionally, I am unable to compile certain libraries, such as pkg-config, for Android.
can we get a script to handle downloading and compilation of libraies OR which libraries and version to download for creating script so it can help freerdp community.

**Reply by akallabeth** [reply]
@88durgesh read the link above, compilation page lists everything...
also, why do you try to compile `pkg-config`?

**Reply by abappdev** [reply]
[RUN] cd /home/build/GITPROJECT/Test/FreeRDP-3.9.0
[RUN] mkdir -p /home/build/GITPROJECT/Test/FreeRDP-3.9.0/scripts/../build/freerdp-build/arm64-v8a
[RUN] cd /home/build/GITPROJECT/Test/FreeRDP-3.9.0/scripts/../build/freerdp-build/arm64-v8a
[RUN] export ANDROID_NDK=/home/build/Android/Sdk/ndk/25.2.9519653
**[RUN] /home/build/Android/Sdk/cmake/3.22.1/bin/cmake -DANDROID_NDK=/home/build/Android/Sdk/ndk/25.2.9519653 -DANDROID_NATIVE_API_LEVEL=android-21 -DCMAKE_TOOLCHAIN_FILE=/home/build/Android/Sdk/ndk/25.2.9519653/build/cmake/android.toolchain.cmake -DCMAKE_BUILD_TYPE=Release -DFREERDP_EXTERNAL_PATH=/home/build/GITPROJECT/Test/FreeRDP-3.9.0/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs -DWITH_CLIENT_SDL=OFF -DWITH_FDK_AAC=ON -DWITH_SERVER=OFF -DCMAKE_MAKE_PROGRAM=make -DWITH_OPENH264=ON -DWITH_MEDIACODEC=OFF -DWITH_FFMPEG=ON -DWITH_SWCALE=ON -DANDROID_ABI=arm64-v8a -DCMAKE_INSTALL_PREFIX=/home/build/GITPROJECT/Test/FreeRDP-3.9.0/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a -DCMAKE_INSTALL_LIBDIR=. -DCMAKE_PREFIX_PATH=/home/build/GITPROJECT/Test/FreeRDP-3.9.0/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a -DCMAKE_SHARED_LINKER_FLAGS=-L/home/build/GITPROJECT/Test/FreeRDP-3.9.0/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a -DcJSON_DIR=/home/build/GITPROJECT/Test/FreeRDP-3.9.0/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a/cmake/cJSON /home/build/GITPROJECT/Test/FreeRDP-3.9.0/scripts/..**
-- The C compiler identification is Clang 14.0.7
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /home/build/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/linux-x86_64/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
Using CFLAGS -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security 
-- Could NOT find ALSA (missing: ALSA_LIBRARY ALSA_INCLUDE_DIR) 
CMake Warning at cmake/ConfigOptions.cmake:168 (find_package):
  By not providing "FindPulseAudio.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "PulseAudio", but CMake did not find one.

  Could not find a package configuration file provided by "PulseAudio" with
  any of the following names:

    PulseAudioConfig.cmake
    pulseaudio-config.cmake

  Add the installation prefix of "PulseAudio" to CMAKE_PREFIX_PATH or set
  "PulseAudio_DIR" to a directory containing one of the above files.  If
  "PulseAudio" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  CMakeLists.txt:80 (include)


-- Audio Disabled
FREERDP_VERSION=3.9.0
-- Git Revision n/a
-- Found ZLIB: /home/build/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/lib/aarch64-linux-android/21/libz.so (found version "1.2.12") 
-- Performing Test Wno-builtin-macro-redefined
-- Performing Test Wno-builtin-macro-redefined - Success
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - not found
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Performing Test fno-omit-frame-pointer
-- Performing Test fno-omit-frame-pointer - Success
-- Performing Test ABI_SOFTFP_SUPPORTED
-- Performing Test ABI_SOFTFP_SUPPORTED - Success
-- Skipping disabled feature Wayland for Wayland (Wayland client)
-- Finding required feature OpenSSL for cryptography (encryption, certificate validation, hashing functions)
-- Found OpenSSL: /home/build/GITPROJECT/Test/FreeRDP-3.9.0/client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a/libcrypto.so (found version "3.3.1")  
-- Skipping optional feature MbedTLS for cryptography (encryption, certificate validation, hashing functions)
--     Enable feature MbedTLS using "-DWITH_MBEDTLS=ON"
-- Skipping disabled feature PCSC for smart card (smart card device redirection)
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Found FFmpeg: /home/build/GITPROJECT/Test/FreeRDP-3.9.0/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a/lib/libavutil.so;/home/build/GITPROJECT/Test/FreeRDP-3.9.0/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a/lib/libavcodec.so  found components: AVUTIL AVCODEC 
-- Finding optional feature OpenH264 for codec (use OpenH264 library)
-- Found OpenH264: /home/build/GITPROJECT/Test/FreeRDP-3.9.0/client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a/libopenh264.so  
-- Found OpenH264: /home/build/GITPROJECT/Test/FreeRDP-3.9.0/client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a/libopenh264.so
-- Skipping optional feature OpenCL for codec (use OpenCL library)
--     Enable feature OpenCL using "-DWITH_OPENCL=ON"
-- Skipping optional feature GSM for codec (GSM audio codec library)
--     Enable feature GSM using "-DWITH_GSM=ON"
-- Skipping optional feature LAME for codec (lame MP3 audio codec library)
--     Enable feature LAME using "-DWITH_LAME=ON"
-- Skipping optional feature FAAD2 for codec (FAAD2 AAC audio codec library)
--     Enable feature FAAD2 using "-DWITH_FAAD2=ON"
-- Skipping optional feature FAAC for codec (FAAC AAC audio codec library)
--     Enable feature FAAC using "-DWITH_FAAC=ON"
-- Skipping optional feature soxr for codec (SOX audio resample library)
--     Enable feature soxr using "-DWITH_SOXR=ON"
-- Using OpenSSL Version: 3.3.1
using default plugins location
Configured RPATH=$ORIGIN/../.:$ORIGIN/..
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of ssize_t
-- Check size of ssize_t - done
-- Looking for strndup
-- Looking for strndup - found
-- Looking for include file unistd.h
-- Looking for include file unistd.h - found
-- Looking for include file execinfo.h
-- Looking for include file execinfo.h - found
-- Looking for backtrace
-- Looking for backtrace - not found
-- Looking for backtrace_symbols
-- Looking for backtrace_symbols - not found
-- Looking for backtrace_symbols_fd
-- Looking for backtrace_symbols_fd - not found
-- Looking for include file inttypes.h
-- Looking for include file inttypes.h - found
-- Looking for include file stdint.h
-- Looking for include file stdint.h - found
-- Looking for include file stdbool.h
-- Looking for include file stdbool.h - found
-- Looking for include file fcntl.h
^[[A-- Looking for include file fcntl.h - found
-- Looking for include file aio.h
-- Looking for include file aio.h - not found
-- Looking for include file sys/timerfd.h
-- Looking for include file sys/timerfd.h - found
-- Looking for include file sys/filio.h
-- Looking for include file sys/filio.h - not found
-- Looking for include file sys/sockio.h
-- Looking for include file sys/sockio.h - not found
-- Looking for include file syslog.h
-- Looking for include file syslog.h - found
-- Looking for include file sys/select.h
-- Looking for include file sys/select.h - found
-- Looking for include file sys/eventfd.h
-- Looking for include file sys/eventfd.h - found
-- Looking for include file unwind.h
-- Looking for include file unwind.h - found
-- Looking for eventfd_read
-- Looking for eventfd_read - found
-- Looking for strerror_r
-- Looking for strerror_r - found
-- Looking for getlogin_r
-- Looking for getlogin_r - not found
-- Looking for getpwuid_r
-- Looking for getpwuid_r - found
-- Performing Test WINPR_HAVE_TM_GMTOFF
-- Performing Test WINPR_HAVE_TM_GMTOFF - Success
-- Looking for include file poll.h
-- Looking for include file poll.h - found
-- Looking for pthread_mutex_timedlock in pthreads
-- Looking for pthread_mutex_timedlock in pthreads - not found
-- Looking for pthread_mutex_timedlock in pthread
-- Looking for pthread_mutex_timedlock in pthread - not found
-- Looking for pthread_mutex_timedlock
-- Looking for pthread_mutex_timedlock - not found
-- Finding recommended feature OpenSSL for cryptography (encryption, certificate validation, hashing functions)
--     Disable feature OpenSSL using "-DWITH_OPENSSL=OFF"
-- Skipping optional feature MbedTLS for cryptography (encryption, certificate validation, hashing functions)
--     Enable feature MbedTLS using "-DWITH_MBEDTLS=ON"
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find JSONC (missing: JSONC_LIBRARY JSONC_INCLUDE_DIR) 
CMake Warning at winpr/libwinpr/CMakeLists.txt:107 (find_package):
  By not providing "Finduriparser.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "uriparser", but CMake did not find one.

  Could not find a package configuration file provided by "uriparser" with
  any of the following names:

    uriparserConfig.cmake
    uriparser-config.cmake

  Add the installation prefix of "uriparser" to CMAKE_PREFIX_PATH or set
  "uriparser_DIR" to a directory containing one of the above files.  If
  "uriparser" provides a separate development package or SDK, be sure it has
  been installed.


-- Looking for timer_create
-- Looking for timer_create - not found
-- Looking for timer_delete
-- Looking for timer_delete - not found
-- Looking for timer_settime
-- Looking for timer_settime - not found
-- Looking for timer_gettime
-- Looking for timer_gettime - not found
-- Looking for pthread_setschedprio
-- Looking for pthread_setschedprio - not found
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find libsystemd (missing: LIBSYSTEMD_LIBRARY LIBSYSTEMD_INCLUDE_DIR) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find JSONC (missing: JSONC_LIBRARY JSONC_INCLUDE_DIR) 
-- Looking for backtrace
-- Looking for backtrace - not found
-- Looking for backtrace
-- Looking for backtrace - not found
add_library(winpr) []
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find Cairo (missing: CAIRO_LIBRARIES) 
CMake Error at /home/build/Android/Sdk/cmake/3.22.1/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
Call Stack (most recent call first):
  /home/build/Android/Sdk/cmake/3.22.1/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /home/build/Android/Sdk/cmake/3.22.1/share/cmake-3.22/Modules/FindPkgConfig.cmake:99 (find_package_handle_standard_args)
  libfreerdp/CMakeLists.txt:101 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/build/GITPROJECT/Test/FreeRDP-3.9.0/build/freerdp-build/arm64-v8a/CMakeFiles/CMakeOutput.log".
See also "/home/build/GITPROJECT/Test/FreeRDP-3.9.0/build/freerdp-build/arm64-v8a/CMakeFiles/CMakeError.log".
[ERROR] /home/build/Android/Sdk/cmake/3.22.1/bin/cmake -DANDROID_NDK=/home/build/Android/Sdk/ndk/25.2.9519653 -DANDROID_NATIVE_API_LEVEL=android-21 -DCMAKE_TOOLCHAIN_FILE=/home/build/Android/Sdk/ndk/25.2.9519653/build/cmake/android.toolchain.cmake -DCMAKE_BUILD_TYPE=Release -DFREERDP_EXTERNAL_PATH=/home/build/GITPROJECT/Test/FreeRDP-3.9.0/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs -DWITH_CLIENT_SDL=OFF -DWITH_FDK_AAC=ON -DWITH_SERVER=OFF -DCMAKE_MAKE_PROGRAM=make -DWITH_OPENH264=ON -DWITH_MEDIACODEC=OFF -DWITH_FFMPEG=ON -DWITH_SWCALE=ON -DANDROID_ABI=arm64-v8a -DCMAKE_INSTALL_PREFIX=/home/build/GITPROJECT/Test/FreeRDP-3.9.0/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a -DCMAKE_INSTALL_LIBDIR=. -DCMAKE_PREFIX_PATH=/home/build/GITPROJECT/Test/FreeRDP-3.9.0/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a -DCMAKE_SHARED_LINKER_FLAGS=-L/home/build/GITPROJECT/Test/FreeRDP-3.9.0/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a -DcJSON_DIR=/home/build/GITPROJECT/Test/FreeRDP-3.9.0/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a/cmake/cJSON /home/build/GITPROJECT/Test/FreeRDP-3.9.0/scripts/.. retured 1

@88durgesh @akallabeth  same issue here

**Reply by akallabeth** [reply]
@abappdev same to you, read the links posted above.
just  changing random parts of the build configuration and hoping for the best does not work.
